src/libraries/Common/src/System/Net/Security/Unix/SafeFreeCredentials.cs

```diff
@@ -57,19 +51,29 @@ private SafeCredentialReference(SafeFreeCredentials target) : base()
             bool ignore = false;
             target.DangerousAddRef(ref ignore);
             Target = target;
-            SetHandle(new IntPtr(0));   // make this handle valid
         }
 
-        protected override bool ReleaseHandle()
+        public void Dispose()
         {
-            SafeFreeCredentials target = Target;
-            if (target != null)
-            {
-                target.DangerousRelease();
-            }
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
 
+        public void Dispose(bool disposing)
+        {
+            SafeFreeCredentials target = Target;
+            target?.DangerousRelease();
             Target = null!;
-            return true;
         }
+
+        ~SafeCredentialReference()
+        {
+            Dispose(false);
+
+#if DEBUG
+            Debug.Fail("Finalizer should never be called. The SafeCredentialReference should be always disposed manually.");
+#endif
+        }
+
     }
```

Comment:
Why is this code duplicated in two files?  Are there differences here between the Windows and Unix implementations?  They look the same.

Reply:
I'll extract this class into one file